File: mobile/lib/images/style_prompt_restore.dart
/// Exact saved prefixes only. Unknown prompts and edited tag strings stay intact.
({String style, String positive})? restoreSavedStyle(String prompt, Iterable<String> presets) {
  final styles = presets.map((s) => s.trim()).where((s) => s.isNotEmpty).toSet().toList()
    ..sort((a,b) => b.length.compareTo(a.length));
  for (final style in styles) {
    if (prompt == style) return (style: style, positive: '');
    if (!prompt.startsWith(style)) continue;
    final rest = prompt.substring(style.length);
    final delimiter = RegExp(r'^\s*,\s*').firstMatch(rest);
    if (delimiter != null) return (style: style, positive: rest.substring(delimiter.end));
  }
  return null;
}

File: src/style-prompt-restore.ts
/** Only split an exact saved prefix; never infer artist names or split mid-prompt. */
export function restoreSavedStyle<T extends {positivePrompt?: string; stylePrompt?: string}>(
  imported: T, presets: readonly {prompt: string}[],
): T {
  if (imported.stylePrompt !== "" || typeof imported.positivePrompt !== "string") return imported;
  const prompt = imported.positivePrompt;
  const styles = [...new Set(presets.map(p => p.prompt.trim()).filter(Boolean))].sort((a,b)=>b.length-a.length);
  for (const style of styles) {
    if (prompt === style) return {...imported, stylePrompt: style, positivePrompt: ""};
    if (!prompt.startsWith(style)) continue;
    const rest = prompt.slice(style.length);
    if (/^\s*,\s*/.test(rest)) return {...imported, stylePrompt: style, positivePrompt: rest.replace(/^\s*,\s*/, "")};
  }
  return imported;
}
